### DISCUSSION TITLE
Open -> read -> close a buffered file?

### DISCUSSION THREAD

**Discussion by CharlieTap** [axboe/liburing#940] [General]
I understand theres no way to pass data from CQE-> SQE etc so I'd assume this would need to use the fixed file system.
I found some references online to potential calls like IORING_OP_OPENAT2_FIXED_FILE but I guess it didn't make it into io_uring. 

Does anyone know if something like this is planned? Similarly CLOSE_FIXED_FILE?

**Comment by axboe** [comment]
See the man pages, like io_uring_prep_openat_direct.3 and io_uring_prep_close_direct.3

**Comment by CharlieTap** [comment]
For those like me that struggle with this:
The term direct in the io_uring system has nothing to do with direct vs buffered files (i.e. avoiding the os cache), it refers solely to the fixed file private table setup. Come to think of it, it doesn't really make sense as theres no syscall that would target just file descriptors opened in direct mode. 😅
